content/guides/docker-concepts/running-containers/overriding-container-defaults.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Overriding container defaults
+keywords: concepts, build, images, container, docker desktop
+description: This concept page will teach you how to override the container defaults using the `docker run` command.
+---
+
+{{< youtube-embed seY7D7Jx690 >}}
+
+## Explanation
+
+When a Docker container starts, it executes an application or command. The container gets this executable (script or file) from its image’s configuration. Containers come with default settings that usually work well, but you can change them if needed. These adjustments help the container's program run exactly how you want it to.
+
+For example, if you have an existing database container that listens on the standard port and you want to run a new instance of the same database container, then you might want to change the port settings the new container listens on so that it doesn’t conflict with the existing container. Sometimes you might want to increase the memory available to the container if the program needs more resources to handle a heavy workload or set the environment variables to provide specific configuration details the program needs to function properly.
+
+The `docker run` command offers a powerful way to override these defaults and tailor the container's behavior to your liking. The command offers several flags that let you to customize container behavior on the fly.
+
+Here's a few ways you can achieve this.
+
+### Overriding the network ports
+
+Sometimes you might want to use separate database instances for development and testing purposes. Running these database instances on the same port might conflict. You can use the `-p` option in `docker run` to map container ports to host ports, allowing you to run the multiple instances of the container without any conflict.
+
+```console
+$ docker run -d -p HOST_PORT:CONTAINER_PORT postgres
+```
+
+### Setting environment variables
+
+This option sets an environment variable `foo` inside the container with the value `bar`.
+
+```console
+$ docker run -e foo=bar postgres env
```

Comment:
If you look at the initial part, it says "For example, if you have an existing database container that listens on the standard port and you want to run a new instance of the same database container, then you might want to change the port settings the new container listens on so that it doesn’t conflict with the existing container." As we're talking about a database container, it does make sense to pick up a database container rather than Alpine. Changing to alpine would disturb the flow of the explanation. Let me know your thoughts.